src/index.js

```diff
@@ -1,65 +1,65 @@
-import Graceful from 'node-graceful';
-import Logger from './common/logger.js';
-import ElmuServer from './server/elmu-server.js';
-import bootstrapper from './bootstrap/server-bootstrapper.js';
+import url from 'url';
+import path from 'path';
+import parseBool from 'parseboolean';
+import educandu from '@educandu/educandu';
+import Logger from '@educandu/educandu/common/logger.js';
 
 const logger = new Logger(import.meta.url);
 
-Graceful.exitOnDouble = true;
-Graceful.timeout = 10000;
-
-(async function index() {
-
-  let container = null;
-
-  Graceful.on('exit', async (event, signal) => {
-    logger.info(`Received ${signal} - Starting graceful exit process`);
-
-    let hasError = false;
-    if (container) {
-      try {
-        logger.info('Start disposing of container');
-        await bootstrapper.disposeContainer(container);
-        logger.info('Container was sucessfully disposed');
-      } catch (err) {
-        logger.fatal(err);
-        hasError = true;
-      }
-    }
-
-    logger.info(`Graceful exit process has finished ${hasError ? 'with' : 'without'} errors`);
-
-    // eslint-disable-next-line no-process-exit
-    process.exit(hasError ? 1 : 0);
-  });
-
-  process.on('uncaughtException', err => {
-    logger.fatal(err);
-    Graceful.exit(1);
-  });
-
-  try {
-
-    logger.info('Starting application');
-
-    container = await bootstrapper.createContainer();
-    const elmuServer = container.get(ElmuServer);
-
-    logger.info('Starting server');
-    elmuServer.listen((err, port) => {
-      if (err) {
-        logger.fatal(err);
-        Graceful.exit(1);
-      } else {
-        logger.info(`App listening on http://localhost:${port}`);
-      }
-    });
-
-  } catch (err) {
-
-    logger.fatal(err);
-    Graceful.exit(1);
-
-  }
-
-})();
+// eslint-disable-next-line no-process-env
+const precessEnv = process.env;
+
+const thisDir = path.dirname(url.fileURLToPath(import.meta.url));
+
+const env = precessEnv.ELMU_ENV || 'dev';
+
+logger.info('Environment is set to %s', env);
+
+const config = {
+  env,
+  port: Number(precessEnv.ELMU_PORT) || 3000,
+  publicFolders: ['../dist', '../static'].map(x => path.resolve(thisDir, x)),
+  sessionDurationInMinutes: Number(precessEnv.ELMU_SESSION_DURATION_IN_MINUTES) || 60,
+  skipMongoMigrations: parseBool(precessEnv.ELMU_SKIP_DB_MIGRATIONS || false.toString()),
+  skipMongoChecks: parseBool(precessEnv.ELMU_SKIP_DB_CHECKS || false.toString())
+};
+
+if (env === 'dev') {
+  config.redirectToHttps = false;
+  config.redirectToNonWwwDomain = false;
+  config.exposeErrorDetails = true;
+  config.mongoConnectionString = 'mongodb://root:rootpw@localhost:27017/dev-educandu-db?replicaSet=educandurs&authSource=admin';
+  config.cdnEndpoint = 'http://localhost:9000';
+  config.cdnRegion = 'eu-central-1';
+  config.cdnAccessKey = 'UVDXF41PYEAX0PXD8826';
+  config.cdnSecretKey = 'SXtajmM3uahrQ1ALECh3Z3iKT76s2s5GBJlbQMZx';
+  config.cdnBucketName = 'dev-educandu-cdn';
```

Comment:
Also here, don't we use ELMU-dedicated storage (mongoDB, S3) ?

Reply:
This is only for local development